server/container/guice/cassandra-guice/src/main/java/org/apache/james/modules/mailbox/CassandraSessionModule.java

```diff
@@ -101,6 +103,25 @@ CassandraSessionConfiguration getCassandraSessionConfiguration(PropertiesProvide
 
     @Provides
     @Singleton
+    FetchBatchSizes getFetchBatchSizesConfiguration(PropertiesProvider propertiesProvider) {
+        try {
+            PropertiesConfiguration configuration = propertiesProvider.getConfiguration(BATCHSIZES_FILE_NAME);
+            FetchBatchSizes fetchBatchSizes = FetchBatchSizes.builder()
```

Comment:
You are duplicating the logic of setting default value when no value is specified therefore violating the DRY principle.
I would have done something like:
```
FetchBatchSizes fetchBatchSizes = FetchBatchSizes.builder();

Optional.ofNullable(configuration.getInteger("metadata")).ifPresent(fetBatchSizes.metadata);
Optional.ofNullable(configuration.getInteger("headers")).ifPresent(fetBatchSizes.headers);
Optional.ofNullable(configuration.getInteger("body")).ifPresent(fetBatchSizes.body);
Optional.ofNullable(configuration.getInteger("full")).ifPresent(fetBatchSizes.full);

return fetchBatchSizes.build();
```